### PR TITLE
Menu compare

### PR DESCRIPTION
**Pull request by cheddartot**

### CODE REVIEW

**Review by tkorisko — APPROVED** [review]

**Review by tkorisko — APPROVED** [review]

File: app/routes.py [hunk 1]
```diff
@@ -48,14 +48,29 @@ def menu_browse():
     restaurants = utils.get_restaurant_details(res_ids)
 
     return render_template('menu-browse.html', restaurants=restaurants, isAdd=True)
-	
-@app.route('/menu-compare', methods=['GET'])
+
+@app.route('/menu-compare/', methods=['GET', 'POST'])
```
Comment (pgayee694) [inline app/routes.py]: should this be a GET and a POST, or just a POST?

File: app/utils.py [hunk 2]
```diff
@@ -151,9 +151,44 @@ def list_to_string(lst):
...
+        tiny_restaurants = models.UserRestaurant.query.filter_by(user_id=user.id).all()
+        for restaurant in tiny_restaurants:
+            restaurants.add(restaurant)
+    restaurant_ids = set()
+    for restaurant in restaurants:
+        restaurant_ids.add(restaurant.restaurant_id)
+    # restaurants_deats = get_restaurant_details(restaurant_ids)
+    return restaurant_ids
+
+# takes a list of users and returns a set of ids for the intersection of all restaurants those users like.
```
Comment (pgayee694) [inline app/utils.py]: move this into the docstring for the function

File: app/utils.py [hunk 3]
```diff
@@ -151,9 +151,44 @@ def list_to_string(lst):
...
+# returns the first user with the given username, None if the user isn't found.
 def find_user_by_username(username_in):
     return models.User.query.filter_by(username=username_in).first()
 
 def find_friendship(id1, id2):
-    return models.Friends.query.filter_by(friend1_id=id1, friend2_id=id2).first()
\ No newline at end of file
+    return models.Friends.query.filter_by(friend1_id=id1, friend2_id=id2).first()
+
+# takes a list of users and returns a set of ids for all restaurants those users like.
```
Comment (pgayee694) [inline app/utils.py]: move this into the docstring for the function

File: app/utils.py [hunk 4]
```diff
@@ -151,9 +151,44 @@ def list_to_string(lst):
 
     return res
 
-#returns the first user with the given username, None if the user isn't found.
+# returns a list of all the friends (as users) for a given user ID
```
Comment (pgayee694) [inline app/utils.py]: move into docstring for function

**Review by pgayee694 — CHANGES_REQUESTED** [review]

File: app/routes.py (hunk 1 above)
Comment (cheddartot) [reply app/routes.py]: Yeah, it doesn't work without GET.

**Review by pgayee694 — APPROVED** [review]

**Pull request merged**